### ISSUE TITLE
Add Portable Mode

### ISSUE THREAD

**Issue by SmilerRyan** [DIDIRUS4/AstralRinth#42]
Since this is useful for testing purposes i would like to have a "portable" mode, where all profiles etc are forced to be stored in the same folder as the executable if a portable.txt file exists. 

![image](https://github.com/user-attachments/assets/8051ad3e-e2fd-40a5-bc73-55faa53c0a38)

**Comment by SmilerRyan** [comment]
Adding onto this from observations I have made to get closer to this, manually setting `THESEUS_CONFIG_DIR` to `UserData` then launching it with command prompt gets you most of the way there, however these files still get left behind:

C:\Users\User\AppData\Local\AstralRinthApp\EBWebView
C:\Users\User\AppData\Roaming\AstralRinthApp\app-window-state.json

![Windows 11 x64-2024-11-16-06-19-12](https://github.com/user-attachments/assets/a466df6b-d5e6-4a1c-b9b7-05e00116f69f)

Although not optimal, another approach if possible is to somehow set `userprofile` to `.` and then create `appdata\local` and `appdata\roaming` folders on start.

![image](https://github.com/user-attachments/assets/baa9f995-a0c6-42b8-9c82-8c2b138760a5)

**Comment by DIDIRUS4** [comment]
Does it only work on Windows? According to the code, everything looks OK, but on macOS I couldn't call another condition and replace the path with UserData.

**Comment by SmilerRyan** [comment]
I can confirm the latest windows build for me strangely enough works to create the folder but doesen't get to fully launch. (something about tags, is modrinth down right now?)

**Comment by DIDIRUS4** [comment]
![image](https://github.com/user-attachments/assets/ac68e41a-7890-4810-9afe-8834d66117df)

I'll try to replace the staging with the default modrinth API subdomain name. This may resolve minor fetch issues during the launcher boot process.
With this URL, I don't have any errors when pnpm install is launched.

**Comment by DIDIRUS4** [comment]
![image](https://github.com/user-attachments/assets/a76a5110-4e5c-4454-8d58-7ba83eda7bb0)


https://api.modrinth.com/v2/tag/category

**Comment by DIDIRUS4** [comment]
Telling about incomprehensible dependencies, I mean this too, because it breaks the build of the project through CI/CD.

![image](https://github.com/user-attachments/assets/e8c2bc5f-cea6-401b-854b-7f5e4d8ec416)

**Comment by SmilerRyan** [comment]
Yeah, it confused me too when i first tried to build this, but at least this is a fault with them and not the source. if it can't connect in a few seconds i would just expect a fallback to have no online features though instead.

**Comment by DIDIRUS4** [comment]
Yes, now I can install minecraft modpacks and see skins (heads) of licenses and offline accounts. Apparently, the real problem was this dependence on the API...
